mountpoint-s3-crt/src/auth/credentials.rs

```diff
@@ -42,6 +54,25 @@ impl CredentialsProvider {
 
         Ok(Self { inner })
     }
+
+    /// Creates the profile credential provider.
+    pub fn new_profile(allocator: &Allocator, options: &CredentialsProviderProfileOptions) -> Result<Self, Error> {
```

Comment:
We should take ownership of the options here.
```suggestion
    pub fn new_profile(allocator: &Allocator, options: CredentialsProviderProfileOptions) -> Result<Self, Error> {
```

Reply:
that should too